### ISSUE TITLE
Fader Channel stuck on 127 

### ISSUE THREAD

**Issue by TomWhitwell** [TomWhitwell/8mu_Public#1]
This is a are issue, seen normally when a board has been experimented with - multiple different firmware etc. 

Symptoms: 
* One channel - for example Channel 3 - is stuck on 127. 
* Happens randomly, attached to a particular device. 
* Multiple firmware & hardware versions, including rev 1 hardware and 1.0.1 firmware 
* Just after the firmware is updated by uf2 drop, the channel works as normal, until the device is power-cycled, when the channel sticks. 

Insights / attempts to mediate: 

Observations 
* In the editor, the channel appears normal = it's set to an active channel 
* There is no difference in EEPROM file from an effected board and a normal board (checked with diff) 
* This is not a simple hardware issue: Evidence: 1) Board works as normal after uf2 added, before power cycle. 2) Board can do normal analogReads in arduino mode with a test sketch.  
* The channel LED doesn't light, which is expected because the value never changes from 127 

Things tried 
* Resetting firmware using ABCD on launch = no difference 
* Re-install SAMD21 bootloader after erasing flash = no difference 
* Changing calibration settings on analog reads to min=1000 max = 3000 = no difference 
* Re-installing arduino firmware = no difference - after the power cycle 
* Installing Circuit Python & removing setup files = no difference 
* Running storage.erase_filesystem() on Circuit Python to entirely wipe the Flash = no difference  
* Add a routine setting all analog channels mode to OUTPUT = no difference 
* Add 1000ms delay on startup = no difference

**Comment by TomWhitwell** [comment]
Try - check the analogRead process in 8mu.ino 

```
  // Read Faders
  for (int j = 0; j < 8; j++) {
    int thisFader = abs((7 * flip) - j);
    analog[j]->update(analogRead(faders[thisFader]));    
    int temp = analog[j]->getValue();

    D(Serial.print(j));
    D(Serial.print(" "));
    D(Serial.println(temp));

    temp = constrain(temp, faderMin, faderMax);
    temp = map(temp, faderMin, faderMax, 0, (1 << adcResolutionBits) - 1);
    if (!flip) temp = ((1 << adcResolutionBits) - 1 ) - temp;
    currentValue[j] = temp;
  }
```
Produces this in serial monitor: 

```
0 3601
1 3277
2 -2147483648
3 3558
4 3492
5 3046
6 2363
7 2098
```

**Comment by TomWhitwell** [comment]
Ensure that analogRead is performing as expected: 

```
// Read Faders
  for (int j = 0; j < 8; j++) {
    int thisFader = abs((7 * flip) - j);
    int thisReading = analogRead(faders[thisFader]);
    analog[j]->update(thisReading);    
    int temp = analog[j]->getValue();

    D(Serial.print(j));
    D(Serial.print(" "));
    D(Serial.print(thisReading));
    D(Serial.print(" "));
    D(Serial.println(temp));

    temp = constrain(temp, faderMin, faderMax);
    temp = map(temp, faderMin, faderMax, 0, (1 << adcResolutionBits) - 1);
    if (!flip) temp = ((1 << adcResolutionBits) - 1 ) - temp;
    currentValue[j] = temp;
  }
```

```
0 831 858
1 931 958
2 1485 -2147483648
3 2116 2143
4 2505 2531
5 2274 2300
6 2122 2151
7 2274 2298
```

**Comment by TomWhitwell** [comment]
ResponsiveAnalogRead is latest version: 
https://github.com/dxinteractive/ResponsiveAnalogRead 
v1.2.1

**Comment by TomWhitwell** [comment]
Check Raw value is correct: 

```
 // Read Faders
  for (int j = 0; j < 8; j++) {
    int thisFader = abs((7 * flip) - j);
    int thisReading = analogRead(faders[thisFader]);
    analog[j]->update(thisReading);    
    int temp = analog[j]->getValue();
    int tempraw = analog[j]->getRawValue();

    D(Serial.print(j));
    D(Serial.print(" "));
    D(Serial.print(tempraw));
    D(Serial.print(" "));
    D(Serial.println(temp));

    temp = constrain(temp, faderMin, faderMax);
    temp = map(temp, faderMin, faderMax, 0, (1 << adcResolutionBits) - 1);
    if (!flip) temp = ((1 << adcResolutionBits) - 1 ) - temp;
    currentValue[j] = temp;
  }
```

Raw value is correct: 

```
0 829 858
1 927 958
2 723 -2147483648
3 2114 2141
4 2502 2529
5 2273 2300
6 2121 2148
7 2276 2298

```

**Comment by TomWhitwell** [comment]
Error was caused by responsive analog read library - smoothValue was uninitialised - so in come cases it could start at spurious values rather than zero. 
Added a local version of the library with issue corrected
https://github.com/dxinteractive/ResponsiveAnalogRead/issues/34

**Comment by TomWhitwell** [comment]
NB this will be resolved in 8mu firmware 1.0.2
